veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,290 @@
+
+#### Sidecar container
+
+`vhost-user` enables high-performance communication between virtual machines and user-space processes, typically offloading VirtIO emulation to dedicated user-space processes (e.g., DPDK or passt). 
+It improves I/O performance through zero-copy mechanisms and reduced context-switching overhead. Additionally, it is required for seamless TCP migration.
+
+`vhost-user` must be enabled in the DomainXML. Small code changes are required in passt sidecar [code](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/sidecars/network-passt-binding). The format is described [here](https://libvirt.org/formatdomain.html#vhost-user-connection-with-passt-backend).
+
+#### CNI plugin 
+
+The [passt-cni](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/cniplugins/passt-binding) process is called by `Multus` during primary network interface setup/removal. It runs two sysctl commands:
+1. Allow binding to all ports starting from 0.
+2. Set ping group range to virt-launcher's user ID.
+> [!NOTE] 
+> No changes are expected in the `passt` CNI code.
+
+### Build and Deployment
+
+#### Add passt-repair binary to virt-handler image
+
+The `passt-repair` binary will be added to the virt-handler container image installed from the `passt` RPM. 
+
+#### passt CNI plugin build and deployment manifests
+
+The [passt-binding directory](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/cniplugins/passt-binding) currently includes a build system to create the CNI plugin binary, however, a container image build is missing.
+Therefore, a container image build will be added.
+
+In addition, missing deployment manifests will be added:
+- A daemonset manifest which serves to deploy the CNI.
+- A manifest for applying a `network-attachment-definition` (NAD) CR named `netbindingpasst` referencing to the CNI.
+
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: netbindingpasst
+  namespace: {{ .namespace }}
+spec:
+  config: '{ "cniVersion": "0.3.1", "name": "netbindingpasst", "plugins": [{"type": "kubevirt-passt-binding"}]}'
+```
+
+> [!NOTES] 
+> - These artifacts already exist under the IPAM Controller repository and can be copied from their [current location](https://github.com/kubevirt/ipam-extensions/tree/v0.1.10-alpha/passt).
+> - For a more secure and storage efficient installation of `passt-helper`, it can be extracted from the `passt` RPM and installed in the virt-handler container image without the 'passt` binary. 
+
+
+## API Examples
+
+No API changes are expected in kubevirt/kubevirt.
+### passt network binding plugin configuration in KubeVirt CR
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+spec:
+  configuration:
+    network:
+      binding:
+        passt:
+          computeResourceOverhead:
+            requests:
+              memory: 250MiB
```

Comment:
Our present documentation requires 500Mi [1].
Is the new passt version requires less memory?

Same question for the scalability section bellow.

[1] https://kubevirt.io/user-guide/network/net_binding_plugins/passt/#passt-registration

Reply:
Yes, we improved this substantially with https://passt.top/passt/commit/?id=eedc81b6ef552736c4d1d7354837e296af081b57 and a couple of follow-ups. You can assume something around 250 MiB with a recent kernel version and a typical KubeVirt configuration.